tests/map_test.go

```diff
@@ -0,0 +1,80 @@
+package tests
```

Comment:
I didn't see anything that existed like this.  This should give an okay exercise of the map api and built-in functions that operate on maps.  If I understand CI right, these should get run by `gopherjs test`.

Reply:
I think these tests are good.  While most/all of these changes may also be exercised by other (stdlib) tests, I think it's good to have explicit tests, to help pinpoint if a failure ever occurs.

I'm not sure of the best place for them to live, though.  Here is probably as good a place as any, unless @nevkontakte  has another idea.  It's always easy to move tests if we want to in the future.

Reply:
Thanks @tomconnell-wf for adding these tests, historically gopherjs test coverage has been somewhat spotty and relied a lot on standard library tests (like @flimzy said). Standard library tests indeed cover most of language use cases, but if something breaks, it makes for a complicated debugging session. So having specific tests is very valuable and welcome.

Placing tests here is okay too. The `tests` package has gotten somewhat out of hand though, and we should reorganize it into several subpackages, but that's out of scope for this PR.